file-formats/smtg/type/zif_aff_smtg_v1.intf.abap

```diff
@@ -0,0 +1,52 @@
+INTERFACE zif_aff_smtg_v1
+  PUBLIC.
+
+  TYPES:
+    "! <p class="shorttext">General Information</p>
+    "! Combines template header and administrative data.
+    BEGIN OF ty_general_information,
+      "! <p class="shorttext">Template ID</p>
+      "! Unique identifier for the template
+      "! $required
+      template_id          TYPE c LENGTH 60,
+      "! <p class="shorttext">Template Name</p>
+      "! Name of an Email Template
+      template_name        TYPE c LENGTH 60,
+      "! <p class="shorttext">Template Description</p>
+      "! Description of an Email Template
+      template_description TYPE c LENGTH 255,
+      "! <p class="shorttext">CDS View</p>
+      "! The corresponding CDS view name for the template
+      "! $required
+      cds_view             TYPE c LENGTH 30,
+      "! <p class="shorttext">Predelivered</p>
+      "! Indicator if the template is pre-delivered
+      is_predelivered      TYPE abap_bool,
+      "! <p class="shorttext">Subject</p>
+      "! The subject of the email template
+      email_subject        TYPE c LENGTH 1024,
+      "! <p class="shorttext">Automatic Text Generation</p>
+      "! Indicator if the template text body is generated automatically from html body
+      autogenerate                 TYPE abap_bool,
```

Comment:
```suggestion
      is_auto_generated                 TYPE abap_bool,
```
or maybe `is_plaintext_auto_generated`? If you use `is_` for booleans above, it would be nice here too, but this is up to you.